lib/ASTGen/Sources/ASTGen/DeclAttrs.swift

```diff
@@ -984,16 +1166,39 @@ extension ASTGenVisitor {
     )
   }
 
-  func generateCustomAttr(attribute node: AttributeSyntax) -> BridgedCustomAttr? {
-    guard
-      let args = node.arguments?.as(LabeledExprListSyntax.self)?[...]
-    else {
-      // TODO: Diagnose.
-      return nil
+  func generateCustomAttr(attribute node: AttributeSyntax, initContext: inout BridgedPatternBindingInitializer?) -> BridgedCustomAttr? {
+    let type = self.generate(type: node.attributeName)
+
+    let argList: BridgedArgumentList?
+      if let args = node.arguments {
+      guard let args = args.as(LabeledExprListSyntax.self) else {
+        // TODO: Diagnose?
+        return nil
+      }
+
+      if !self.declContext.isLocalContext && initContext == nil {
+        initContext = BridgedPatternBindingInitializer.create(declContext: self.declContext)
+      }
```

Comment:
This reminds me, currently we create these even when attached to non-PatternBindingDecls :( This crashes (rdar://100585434):

```swift
@attached(member)
macro M<T>(_ x: T) = #externalMacro(module: "", type: "")

@M({})
struct S {}
```

Not to be solved in this PR though 😄

Reply:
Huh. Do we only need it for PBD? If so, I think we should just pre-create `PatternBindingInitializer` when generating `PatternBindingDecl` if it has any attributes or initializers, and generate the attributes and the initializers with `withDeclContext`

Reply:
I believe we need _some_ local context to contextualize closures for the non-PatternBindingDecl cases, but it can't be PatternBindingInitializer, we probably need a new kind of context (or maybe we could rework PatternBindingInitializer such that it doesn't assume it's on a PBD).